File: src/main/java/com/extendedae_plus/common/registry/menu/host/linkLabel/HostLabelLink.java
package com.extendedae_plus.common.registry.menu.host.linkLabel;

import com.extendedae_plus.common.wireless.linkApi.Label;

public interface HostLabelLink {
    Label.Data getLabelData();

    boolean setLabelData(Label.Data label, boolean force);

    default boolean isLockable() {
        return false;
    }

    default boolean isMasterable() {
        return false;
    }

    default boolean isLocked() {
        return false;
    }

    default boolean isMaster() {
        return false;
    }

    default void toggleLock() {
    }

    default void toggleMaster() {
    }
}

File: src/main/kotlin/com/extendedae_plus/common/registry/menu/host/link/HostLabelLink.kt
package com.extendedae_plus.common.registry.menu.host.link

import com.extendedae_plus.common.wireless.linkApi.Label

interface HostLabelLink {
    val labelData: Label.Data

    fun setLabelData(label: Label.Data, force: Boolean = false): Boolean

    val isLockable: Boolean
        get() = false

    val isMasterable: Boolean
        get() = false

    val isLocked: Boolean
        get() = false

    val isMaster: Boolean
        get() = false

    fun toggleLock() {
    }

    fun toggleMaster() {
    }
}
